crates/router/src/connector/klarna/transformers.rs

```diff
@@ -240,53 +369,125 @@ fn get_address_info(
     })
 }
 
-impl TryFrom<types::PaymentsResponseRouterData<KlarnaPaymentsResponse>>
+impl TryFrom<types::PaymentsResponseRouterData<KlarnaAuthResponse>>
     for types::PaymentsAuthorizeRouterData
 {
     type Error = error_stack::Report<errors::ConnectorError>;
+
     fn try_from(
-        item: types::PaymentsResponseRouterData<KlarnaPaymentsResponse>,
+        item: types::PaymentsResponseRouterData<KlarnaAuthResponse>,
     ) -> Result<Self, Self::Error> {
         let connector_response = types::ConnectorResponseData::with_additional_payment_method_data(
-            match item.response.authorized_payment_method {
-                Some(authorized_payment_method) => {
-                    types::AdditionalPaymentMethodConnectorResponse::from(authorized_payment_method)
+            match item.response {
+                KlarnaAuthResponse::KlarnaPaymentsAuthResponse(ref response) => {
+                    match &response.authorized_payment_method {
+                        Some(authorized_payment_method) => {
+                            types::AdditionalPaymentMethodConnectorResponse::from(
+                                authorized_payment_method.clone(),
+                            )
+                        }
+                        None => types::AdditionalPaymentMethodConnectorResponse::PayLater {
+                            klarna_sdk: None,
+                            klarna_checkout: None,
+                        },
+                    }
                 }
-                None => {
-                    types::AdditionalPaymentMethodConnectorResponse::PayLater { klarna_sdk: None }
+                KlarnaAuthResponse::KlarnaCheckoutAuthResponse(ref response) => {
+                    match &response.authorized_payment_method {
+                        Some(authorized_payment_method) => {
+                            types::AdditionalPaymentMethodConnectorResponse::from(
+                                authorized_payment_method.clone(),
+                            )
+                        }
+                        None => types::AdditionalPaymentMethodConnectorResponse::PayLater {
+                            klarna_sdk: None,
+                            klarna_checkout: None,
+                        },
+                    }
                 }
             },
         );
+        let payment_method_data = item.data.request.payment_method_data.clone();
+
+        match payment_method_data {
```

Comment:
You can match directly on the response type, similar to how you handled the psync response.